shell/platform/darwin/ios/framework/Source/platform_views_controller.mm

```diff
@@ -104,71 +104,223 @@ bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
+               surfaceFrames:
+                   (const std::vector<std::unique_ptr<flutter::SurfaceFrame>>&)surfaceFrames;
+- (void)onCreate:(FlutterMethodCall*)call result:(FlutterResult)result;
+- (void)onDispose:(FlutterMethodCall*)call result:(FlutterResult)result;
+- (void)onAcceptGesture:(FlutterMethodCall*)call result:(FlutterResult)result;
+- (void)onRejectGesture:(FlutterMethodCall*)call result:(FlutterResult)result;
+
+- (void)clipViewSetMaskView:(UIView*)clipView;
+
+// Applies the mutators in the mutatorsStack to the UIView chain that was constructed by
+// `ReconstructClipViewsChain`
+//
+// Clips are applied to the `embeddedView`'s super view(|ChildClippingView|) using a
+// |FlutterClippingMaskView|. Transforms are applied to `embeddedView`
+//
+// The `boundingRect` is the final bounding rect of the PlatformView
+// (EmbeddedViewParams::finalBoundingRect). If a clip mutator's rect contains the final bounding
+// rect of the PlatformView, the clip mutator is not applied for performance optimization.
+//
+// This method is only called when the `embeddedView` needs to be re-composited at the current
+// frame. See: `compositeView:withParams:` for details.
+- (void)applyMutators:(const flutter::MutatorsStack&)mutatorsStack
+         embeddedView:(UIView*)embeddedView
+         boundingRect:(const SkRect&)boundingRect;
+// Appends the overlay views and platform view and sets their z index based on the composition
+// order.
+- (void)bringLayersIntoView:(const LayersMap&)layerMap
+       withCompositionOrder:(const std::vector<int64_t>&)compositionOrder;
+- (std::shared_ptr<flutter::OverlayLayer>)nextLayerInPool;
+- (void)createLayerWithIosContext:(const std::shared_ptr<flutter::IOSContext>&)iosContext
+                        grContext:(GrDirectContext*)grContext
+                      pixelFormat:(MTLPixelFormat)pixelFormat;
+- (void)removeUnusedLayers:(const std::vector<std::shared_ptr<flutter::OverlayLayer>>&)unusedLayers
+      withCompositionOrder:(const std::vector<int64_t>&)compositionOrder;
+- (std::vector<UIView*>)uiViewsToDispose;
+- (void)resetFrameState;
+@end
+
 namespace flutter {
 
+// TODO(cbracken): Eliminate the use of globals.
 // Becomes NO if Apple's API changes and blurred backdrop filters cannot be applied.
 BOOL canApplyBlurBackdrop = YES;
 
-PlatformViewsController::PlatformViewsController()
-    : layer_pool_(std::make_unique<OverlayLayerPool>()),
-      weak_factory_(std::make_unique<fml::WeakPtrFactory<PlatformViewsController>>(this)) {
-  mask_view_pool_ =
-      [[FlutterClippingMaskViewPool alloc] initWithCapacity:kFlutterClippingMaskViewPoolCapacity];
-};
+}  // namespace flutter
 
-void PlatformViewsController::SetTaskRunner(
-    const fml::RefPtr<fml::TaskRunner>& platform_task_runner) {
-  platform_task_runner_ = platform_task_runner;
-}
+@implementation FlutterPlatformViewsController {
+  std::unique_ptr<flutter::OverlayLayerPool> _layerPool;
```

Comment:
can you add a comment on why we have to keep these separate ivars (for returning the ref rather than copy?)

Reply:
Done.